File: ones-gfx-sdk-go/ones_gfx/resources/peering.go
package resources

import (
	"context"
	"fmt"

	"github.com/aviznetworks/ones-gfx-sdk/ones-gfx-sdk-go/ones_gfx"
)

// PeeringResource handles VPC peering operations.
type PeeringResource struct {
	transport *ones_gfx.Transport
}

// NewPeeringResource constructs a PeeringResource.
func NewPeeringResource(transport *ones_gfx.Transport) *PeeringResource {
	return &PeeringResource{transport: transport}
}

// Create creates a peering between two VPCs.
// Maps to POST /fabrics/{fabricName}/vpcpeering.
func (r *PeeringResource) Create(ctx context.Context, fabricName, name, vpcName, peerVPCName string) (interface{}, error) {
	if fabricName == "" {
		return nil, fmt.Errorf("fabricName is required")
	}
	if name == "" {
		return nil, fmt.Errorf("name is required")
	}
	if vpcName == "" {
		return nil, fmt.Errorf("vpcName is required")
	}
	if peerVPCName == "" {
		return nil, fmt.Errorf("peerVPCName is required")
	}

	body := map[string]string{
		"name":        name,
		"vpcname":     vpcName,
		"peervpcname": peerVPCName,
	}

	path := fmt.Sprintf("fabrics/%s/vpcpeering", fabricName)
	result, err := r.transport.Post(path, body, ones_gfx.OperationModeSynchronous, nil)
	if err != nil {
		return nil, err
	}

	return result, nil
}

File: ones-gfx-sdk-python/ones_gfx/resources/peering.py
from __future__ import annotations

from typing import Any

from ..transport import Transport


class PeeringResource:
    def __init__(self, transport: Transport):
        self._transport = transport

    def create(self, fabric_name: str, name: str, vpc_name: str, peer_vpc_name: str) -> Any:
        if not fabric_name:
            raise ValueError("fabric_name is required.")
        if not name:
            raise ValueError("name is required.")
        if not vpc_name:
            raise ValueError("vpc_name is required.")
        if not peer_vpc_name:
            raise ValueError("peer_vpc_name is required.")
        body = {"name": name, "vpcname": vpc_name, "peervpcname": peer_vpc_name}
        return self._transport.post(f"fabrics/{fabric_name}/vpcpeering", json_body=body)
